SwrveSDK/src/main/java/com/swrve/sdk/SwrveBase.java

```diff
@@ -179,15 +184,35 @@ protected T init(final Activity activity) throws IllegalArgumentException {
                 // Add custom message listener
                 if (messageListener == null) {
                     setMessageListener(new ISwrveMessageListener() {
-                        public void onMessage(final SwrveMessage message, boolean firstTime) {
-                            if (SwrveBase.this.activityContext != null) {
-                                final Activity activity = SwrveBase.this.activityContext.get();
-                                if (activity == null) {
-                                    SwrveLogger.e(LOG_TAG, "Can't display a message with a non-Activity context");
+                        public void onMessage(final SwrveMessage message) {
```

Comment:
Can we reduce the number of lines in the init method and refactor so that this logic is in its own method. Move the whole setMessageListener.